packages/ipfs-core/src/components/id.js

```diff
@@ -14,44 +16,71 @@ module.exports = ({ peerId, libp2p }) => {
   /**
    * Returns the identity of the Peer
    *
-   * @param {import('../utils').AbortOptions} [_options]
+   * @param {IdOptions} [options]
    * @returns {Promise<PeerId>}
    * @example
    * ```js
    * const identity = await ipfs.id()
    * console.log(identity)
    * ```
    */
-  async function id (_options) { // eslint-disable-line require-await
-    const id = peerId.toB58String()
+  async function id (options = {}) { // eslint-disable-line require-await
+    options = options || {}
+
+    let id = peerId
+    let publicKey = id.pubKey
     let addresses = []
     let protocols = []
+    let agentVersion = `js-ipfs/${pkgversion}`
+    let protocolVersion = '9000'
+
+    if (options.peerId) {
+      if (PeerId.isPeerId(options.peerId)) {
+        id = options.peerId
+      } else {
```

Comment:
NIT: do we really need to support different representations ? Seems like user of the API could do the coercion instead.

Reply:
No, we don't.  Also, this should be a string and not a PeerId, otherwise we pull `peer-id` and all the associated crypto junk into `ipfs-core-types` as it'll form part of the API types and so into the http client and other places that don't need them.